src/lib/vault/getNewFormSubmissions.ts

```diff
@@ -0,0 +1,60 @@
+import { QueryCommand, type QueryCommandOutput } from "@aws-sdk/lib-dynamodb";
+import { AwsServicesConnector } from "@lib/awsServicesConnector";
+import {
+  newFormSubmissionFromDynamoDbResponse,
+  type NewFormSubmission,
+} from "@lib/vault/dataStructures/formSubmission";
+
+export async function getNewFormSubmissions(
+  formId: string,
+  limit: number,
+): Promise<NewFormSubmission[]> {
+  try {
+    let newFormSubmissions: NewFormSubmission[] = [];
+    let lastEvaluatedKey: Record<string, string> | undefined | null = null;
+
+    while (lastEvaluatedKey !== undefined) {
+      const response: QueryCommandOutput =
+        await AwsServicesConnector.getInstance().dynamodbClient.send(
+          new QueryCommand({
+            TableName: "Vault",
+            IndexName: "StatusCreatedAt",
+            ExclusiveStartKey: lastEvaluatedKey ?? undefined,
+            Limit: limit - newFormSubmissions.length,
```

Comment:
Can you explain to me again why we would need the evaluated key to ensure the limit works properly?

Reply:
Even though this may not be useful just now because of the maximum number of new form submission we are requesting and the way DynamoDB handle pagination..

> A single Query operation will read up to the maximum number of items set (if using the Limit parameter) or a maximum of 1 MB of data and then apply any filtering to the results using FilterExpression. If LastEvaluatedKey is present in the response, you will need to paginate the result set. For more information, see [Paginating the Results ](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.html#Query.Pagination)
 in the Amazon DynamoDB Developer Guide.

 .. I wanted to have this logic built in for if something changes on their side or ours (we increase our maximum of returned new form submissions or we have to request more fields for each entry).